File: inject.js
const fs = require('fs');
const path = require('path');

const files = fs.readdirSync(__dirname).filter(f => f.endsWith('.html') && !f.startsWith('admin') && !f.startsWith('dynamic'));

files.forEach(f => {
    const filePath = path.join(__dirname, f);
    const content = fs.readFileSync(filePath, 'utf8');

    if (!content.includes('kip-frontend.js')) {
        let newContent = content.replace('<script src="script.js"></script>', '<script src="admin-data.js"></script>\n    <script src="kip-frontend.js"></script>\n    <script src="script.js"></script>');

        if (newContent === content) {
            newContent = content.replace('</body>', '    <script src="admin-data.js"></script>\n    <script src="kip-frontend.js"></script>\n</body>');
        }

        if (newContent !== content) {
            fs.writeFileSync(filePath, newContent, 'utf8');
            console.log('Updated ' + f);
        }
    }
});

File: inject.py
import glob

files = glob.glob('*.html')
for f in files:
    if f.startswith('admin') or f.startswith('dynamic'):
        continue
    
    with open(f, 'r', encoding='utf-8') as file:
        content = file.read()
        
    if 'kip-frontend.js' not in content:
        new_content = content.replace('<script src="script.js"></script>', '<script src="admin-data.js"></script>\n    <script src="kip-frontend.js"></script>\n    <script src="script.js"></script>')
        
        if new_content == content:
            new_content = content.replace('</body>', '    <script src="admin-data.js"></script>\n    <script src="kip-frontend.js"></script>\n</body>')
            
        if new_content != content:
            with open(f, 'w', encoding='utf-8') as file:
                file.write(new_content)
            print(f'Updated {f}')
